[v9] `tsh` list resources accross proxies and clusters (#12934)

atburke opened the pull request:
Backport of #12934.

atburke committed:
`tsh` list resources accross proxies and clusters (#12934)

This change adds the --all/-R flag to tsh ls, tsh apps ls, tsh db ls, and tsh kube ls, which lets tsh list resources from across all clusters and logged in proxies.

atburke committed:
Remove unused formatKubeLabels

atburke committed:
Merge branch 'branch/v9' into atburke/v9/tsh-ls-across-clusters

jakule reviewed (changes requested):
I think this PR needs to wait for https://github.com/gravitational/teleport/pull/13321 as currently master doesn't work.

atburke committed:
Fix dependencies in integration tests (#13321)

This change moves some type/function definitions in integration tests to fix compilation.

atburke committed:
Merge branch 'branch/v9' into atburke/v9/tsh-ls-across-clusters

atburke committed:
Fix NewRole

atburke committed:
Merge branch 'branch/v9' into atburke/v9/tsh-ls-across-clusters

atburke committed:
Fix settesttimeouts again

atburke committed:
Merge branch 'branch/v9' into atburke/v9/tsh-ls-across-clusters
